sdks/python/apache_beam/transforms/window.py

```diff
@@ -341,14 +343,18 @@ def __ne__(self, other):
 
   def to_runner_api_parameter(self, context):
     return (urns.FIXED_WINDOWS_FN,
-            proto_utils.pack_Struct(size=self.size.micros,
-                                    offset=self.offset.micros))
-
-  @urns.RunnerApiFn.register_urn(urns.FIXED_WINDOWS_FN, struct_pb2.Struct)
+            standard_window_fns_pb2.FixedWindowsPayload(
+                size=proto_utils.from_micros(
+                    duration_pb2.Duration, self.size.micros),
+                offset=proto_utils.from_micros(
+                    timestamp_pb2.Timestamp, self.offset.micros)))
+
+  @urns.RunnerApiFn.register_urn(
```

Comment:
I generally think that two-way translation is better via separate registered translators than dynamic dispatch. So that is where the design comes from. But actually, we don't _really_ do it, for a specific reason. When we are parsing a proto that has URN "beam:transforms:pardo" we don't try to make a Java `ParDo` object because the types wouldn't work out, etc. Instead all transforms just become `RawPTransform` that can vend its URN and language-agnostic bits of the payload. I am currently fleshing out the kinds of access runners will need.

Just notes, nothing particularly needed here.